src/main/scala/io/scalac/tezos/translator/model/types/ContactData.scala

```diff
@@ -0,0 +1,49 @@
+package io.scalac.tezos.translator.model.types
+
+import eu.timepit.refined.W
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.boolean.{And, Not}
+import eu.timepit.refined.collection.{NonEmpty, Size}
+import eu.timepit.refined.numeric.Greater
+import eu.timepit.refined.string.MatchesRegex
+import io.circe.{Decoder, Encoder}
+import io.estatico.newtype.macros.newtype
+import io.scalac.tezos.translator.model.types.Util._
+import sttp.tapir.{Schema, SchemaType}
+
+object ContactData {
+
+  type NameAndEmailReq = NonEmpty And Size[Not[Greater[W.`255`.T]]]
```

Comment:
The purpose of refined types its guarantee that we have only the right data for a field, it's opposed to validation because we decide to drop that data on parsing level, and with it, we wanna remove validation and have the guarantee on the type level. With validation, there is no guarantee in types that email and name is valid, but with refined types - yes

Reply:
And yes, it's not a validation of email because I don't have regex here, so about separate you are right, and it's will be good to rename email field of course or add regex